CONTRIBUTING.md

```diff
@@ -0,0 +1,79 @@
+```
+mkdir build && cd build
+cmake .. && cmake --build . --config Release
+```
+
+3. Make sure the tests pass:
+```
+make tests			# Application tests
+./bin/unit_tests	# Unit tests
+```
+
+4. Make your change. Add tests for your change. Make the tests pass.
+5. Create the Pull Request (PR).
+	a. Suggest a reviewer if you know that some of the mantainers know about the specific change.
+    b. Check if the CI checks pass.
+    c. If your PR lives for a long time, try to avoid to press the button "Update branch" in the Pull Request view. We prefer rebasing in top of master, instead of merging master into branches.
+
+## Pull Request guidelines
+- PRs should be kept at a manageable size. Try to focus in just one goal per PR. If you find yourself doing several things that in a PR that were not expected, try to split the different tasks in different PRs.
+- Commits should always change one logical unit, so that cherry-picking & reverting is simple.
+- Commit messages should as informative and concise as possible. The first line of the commit message should be < 80 characters and describe the commit briefly. If the 80 characters are too short for a summary, consider splitting the commit. Below the short summary add one blank line and then a more detailed explanation if required.
+
+## Code guidelines ##
+### General ###
+- All new code must be properly tested: via unit tests (based on the Gtest framework; see `$REPO/unitTest`) or system tests (scripts exercising the main exiv2 application; see `$REPO/test`).
+- Code should be simple to read and to understand.
+- Do not invoke undefined behavior! Ensure that with UBSAN, i.e. compile your code with `-fsanitize=undefined` and run the test suite.
+- Ensure that your code has no memory errors. Use ASAN for that, i.e. compile your code with `-fsanitize=address`.
+
+### Relevant to the Exiv2 domain ###
+- All new code that is added must be resistant to integer overflows, thus if you multiply, add, subtract, divide or bitshift integers you must ensure that no overflow can occur. Please keep in mind that signed integer overflow is undefined behavior, thus you must check for overflows before performing the arithmetic operation, otherwise the compiler is free to optimize your check after the overflow away (this has happened already).
+- All new code must be resistant to buffer overflows. Thus before you access arrays a range check must be performed.
+- Distrust any data that you extract from images or from external sources. E.g. if the metadata of an image gives you an offset of another information inside that file, do not assume that this offset will not result in an out off bounds read.
+- New code must not assume the endianes and the wordsize of the system it is being run on. I.e. don't assume that  `sizeof(int) = 8` or that the following will work:
+```cpp
+const uint32_t some_var = get_var();
+const uint16_t lower_2_bytes = (const uint16_t*) &some_var;
+```
```

Comment:
As a site note: Detecting integer overflows in a C++ portable way is pain. I recently dicovered https://github.com/nemequ/portable-snippets/tree/master/safe-math which I intent to use in my other projects.

Reply:
I know, especially if you intend to support other compilers than gcc and clang which provide very efficient builtins for that. I intend to provide something portable and usable in the near future.